lib/internal/url.js

```diff
@@ -196,16 +188,12 @@ function onParsePathComplete(flags, protocol, username, password,
 
 function onParseSearchComplete(flags, protocol, username, password,
                                host, port, path, query, fragment) {
-  if (flags & binding.URL_FLAGS_FAILED)
-    return;
```

Comment:
Yes, the completion and the errors are handled in different callbacks, since it doesn't really make sense to pass all these parsed parts to an error callback anyway.

The setters are ignoring errors per the spec at the moment, so they don't have error callbacks.